spec/operators/multicast-spec.ts

```diff
@@ -702,6 +702,22 @@ describe('multicast operator', () => {
   });
 
   // TODO: fix firehose unsubscription
+  // AFAICT, it's not possible for multicast observables to support ASAP
+  // unsubscription from synchronous firehose sources. The problem is that the
+  // chaining of the closed 'signal' is broken by the subject. For example,
+  // here:
+  //
+  // https://github.com/ReactiveX/rxjs/blob/2d5e4d5bd7b684a912485e1c1583ba3d41c8308e/src/internal/operators/multicast.ts#L53
+  //
+  // The subject is passed to subscribe. However, in the subscribe
+  // implementation a SafeSubcriber is created with the subject as the
+  // observer:
+  //
+  // https://github.com/ReactiveX/rxjs/blob/2d5e4d5bd7b684a912485e1c1583ba3d41c8308e/src/internal/Observable.ts#L210
+  //
+  // That breaks the chaining of closed - i.e. even if the unsubscribe is
+  // called on the subject, closing it, the SafeSubscriber's closed property
+  // won't reflect that.
```

Comment:
FYI, `multicast`-based operators are not going to pass the synchronous firehose test with the current implementation of the `closed` 'signal'.

Reply:
I suppose this could be fixed by bringing back the [`SubjectSubscriber`](https://github.com/ReactiveX/rxjs/blob/5c68e208512dd0f3ba1aafbf3e592592ec13db9b/src/internal/Subject.ts#L13-L17).